.github/workflows/php.yml

```diff
@@ -0,0 +1,28 @@
+name: php
+
+on:
+  workflow_call:
+    inputs:
+      php:
+        default: "8.3"
+        type: string
+
+jobs:
+  php:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ inputs.php }}
+          tools: composer:v2
+          extensions: json, dom, curl, libxml, mbstring
+          coverage: none
+
+      - name: Install Composer
+        uses: nick-fields/retry@v3
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          command: composer install --no-interaction --no-progress
```

Comment:
Doesn't need to be done now, but making a mental note. We'll want a way to indicate if we should run `composer install` or `composer update`. The reason is, for most repositories, `composer install` is what we run, but if we start adding our own packages, we'll want to run `composer update` instead.